### PR TITLE
Flooding model

### PR DESCRIPTION
**Pull request by AloysTt**
This PR adds support for the flooding model.
The model is used for the <15m stratum, if there are enough wells. The region will be widened up to 5km if there are not enough wells.

### CODE REVIEW

**Review by TxJson — APPROVED** [review]
Looks good, please rebase into flooding

**Comment by TxJson** [comment]
> Looks good, please rebase into flooding

Ignore this comment, was a misunderstanding.

File: preprocessing/models/model5-estimator.js [hunk 1]
```diff
@@ -97,28 +97,36 @@ function produceEstimate(divisions, div, dis, upa, uni, depth, colour, utensil)
...
+      retval = createMessage(arsenicValues.m_p75);
     }
   } else {
-    retval.message = 'We are unable to assess your tubewell with the information you supplied, please fill all the sections';
+    // regular model
+    if (colour === 'Black' || utensil === 'No colour change to slightly blackish') {
+      const warningSeverity = (depth > 150) ? 'HIGHLY ' : '';
+
+      const floodWarning =
+        (depth <= 15.3 && flood === 'no')
```
Comment (jacekkopecky) [inline preprocessing/models/model5-estimator.js:115]: check everywhere that we're consistent with <15.3 or <=15.3 and make it consistent

File: preprocessing/models/model5-estimator.js [hunk 2]
```diff
@@ -97,28 +97,36 @@ function produceEstimate(divisions, div, dis, upa, uni, depth, colour, utensil)
...
+          ? ' but may be vulnerable to nitrate and pathogens'
```
Comment (jacekkopecky) [inline preprocessing/models/model5-estimator.js:116]: check with Mo: when we are in under 50ft and don't have enough data for the flooding model, whether we should warn that the well "may be vulnerable to nitrate and pathogens" and should we do that if the user said it gets flooding, or if the user said it didn't?

File: preprocessing/models/model5-preprocessor.js [hunk 3]
```diff
@@ -53,9 +54,12 @@ including pre-processed arsenic concentration data which looks like this:
               {
                 union: '..',
                 s15: {
-                  m: ...,   // short for message ID
-                  l: ...,   // short for lower quantile
-                  u: ...,   // short for upper quantile
+                  m: ...,     // short for message ID using median
+                  l: ...,     // short for lower quantile
+                  u: ...,     // short for upper quantile
+                  m_p25: ..., // message ID using the 25th percentile
```
Comment (jacekkopecky) [inline preprocessing/models/model5-preprocessor.js:60]: use m2, m7, m9 for the names here please

File: preprocessing/models/model5-preprocessor.js [hunk 4]
```diff
@@ -153,8 +157,18 @@ function computeWellStats(locationArr) {
   for (const stratum of STRATA) {
     let wells = location[stratum];
 
+    let useFloodingModel = false;
+    if (stratum === 's15') {
+      if (!isEnoughData(wells)) {
+        const wellsWithin5Km = nearbyLocations(locationArr, 5).map(strataSelector(['s15']));
```
Comment (jacekkopecky) [inline preprocessing/models/model5-preprocessor.js:163]: move widening into getEnoughData, possibly add a new stratum like s15Flooding

**Review by jacekkopecky — CHANGES_REQUESTED** [review]
thank you!

File: preprocessing/models/model5-estimator.js (hunk 2 above)
Comment (AloysTt) [reply preprocessing/models/model5-estimator.js:119]: He said that we should not do that as we do not have data for that. I removed it.

File: preprocessing/models/model5-preprocessor.js [hunk 5]
```diff
@@ -322,6 +346,11 @@ function extractStats(data, hierarchyPath) {
           l: dataObj[`${stratum}_low`],
           u: dataObj[`${stratum}_upp`],
         };
+        if (`${stratum}_p25` in dataObj) {
```
Comment (jacekkopecky) [inline preprocessing/models/model5-preprocessor.js:349]: below, please use `.m2` instead of `['m2']`, and same for 7 and 9

**Review by jacekkopecky — CHANGES_REQUESTED** [review]
I've only reviewed a bit of it, sorry, some comments are below

**Comment by jacekkopecky** [comment]
depends on #44 which should be merged first, then this work may need to be ported over to add mouzas

**Comment by jacekkopecky** [comment]
redone in #58 and merged

**Pull request closed without merging**